asllib/Typing.ml

```diff
@@ -2237,24 +2362,28 @@ module Annotate (C : ANNOTATE_CONFIG) : S = struct
               let t = T_Bits (width, []) |> here in
               check_type_satisfies le env t_e t ()
             in
-            let slices2 = best_effort slices (annotate_slices env) in
+            let slices2, ses2 =
+              best_effort (slices, SES.empty) @@ fun _ ->
+              annotate_slices env slices ~loc
+            in
             let+ () = check_disjoint_slices le env slices2 in
             let+ () =
               check_true (not (list_is_empty slices)) @@ fun () ->
               fatal_from le Error.EmptySlice
             in
-            LE_Slice (le2, slices2) |> here |: TypingRule.LESlice
+            let ses = SES.union ses1 ses2 in
```

Comment:
I think the semantics evaluates the base of the slice and the slices in parallel so probably you should use `conflicting_union`, no?